### PR TITLE
docs(prometheus) add new options for 3.0

### PR DESCRIPTION
**Pull request by fffonion**
### Summary
docs(prometheus) add new options for 3.0

### Reason
From https://github.com/Kong/kong/pull/9028

### CODE REVIEW

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+---
+name: Prometheus
+publisher: Kong Inc.
+version: 1.6.0
+desc: Expose metrics related to Kong and proxied Upstream services in Prometheus exposition format
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:5]:
```suggestion
desc: Expose metrics related to Kong and proxied upstream services in Prometheus exposition format
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+description: |
+  Expose metrics related to Kong and proxied Upstream services in [Prometheus](https://prometheus.io/docs/introduction/overview/) exposition format, which can be scraped by a Prometheus Server.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:7]:
```suggestion
  Expose metrics related to Kong and proxied upstream services in [Prometheus](https://prometheus.io/docs/introduction/overview/) exposition format, which can be scraped by a Prometheus Server.
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      default: '`false`'
+      description: |
+        A boolean value that determines if per-consumer metrics should be
+        collected.
+        If enabled, a `kong_http_consumer_status` metric is added to
+        exported metrics.
+---
+
+Metrics tracked by this plugin are available on both the Admin API and Status
+API at the `http://localhost:<port>/metrics`
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:47]:
```suggestion
API at the `http://localhost:PORT/metrics`
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+endpoint. Note that the URL to those APIs will be specific to your
+installation; see [Accessing the metrics](#accessing-the-metrics).
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:49]:
```suggestion
installation, see [Accessing the metrics](#accessing-the-metrics).
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+This plugin records and exposes metrics at the node level. Your Prometheus
+server will need to discover all Kong nodes via a service discovery mechanism,
+and consume data from each node's configured `/metrics` endpoint.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:53]:
```suggestion
This plugin records and exposes metrics at the node level. Your Prometheus
server must discover all Kong nodes via a service discovery mechanism
and consume data from each node's configured `/metrics` endpoint.
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+## Grafana dashboard
+
+Metrics exported by the plugin can be graphed in Grafana using a drop in
+dashboard: [https://grafana.com/dashboards/7424](https://grafana.com/dashboards/7424).
+
+## Available metrics
+
+- **Status codes**: HTTP status codes returned by Upstream services.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:62]:
```suggestion
- **Status codes**: HTTP status codes returned by upstream services.
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+  These are available per service, across all services, and per route per consumer.
+- **Latencies Histograms**: Latency in ms, as measured at Kong:
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:64]:
```suggestion
- **Latencies Histograms**: Latency (in ms), as measured at Kong:
```

File: app/_hub/kong-inc/prometheus/1.6.0.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+   - **Request**: Total time taken by Kong and Upstream services to serve
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/1.6.0.md:65]:
```suggestion
   - **Request**: Total time taken by Kong and upstream services to serve
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
 ---
 name: Prometheus
 publisher: Kong Inc.
-version: 1.6.0
+version: 3.0.0
 desc: Expose metrics related to Kong and proxied Upstream services in Prometheus exposition format
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:5]:
```suggestion
desc: Expose metrics related to Kong and proxied upstream services in Prometheus exposition format
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
 description: |
   Expose metrics related to Kong and proxied Upstream services in [Prometheus](https://prometheus.io/docs/introduction/overview/) exposition format, which can be scraped by a Prometheus Server.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:7]:
```suggestion
  Expose metrics related to Kong and proxied upstream services in [Prometheus](https://prometheus.io/docs/introduction/overview/) exposition format, which can be scraped by a Prometheus Server.
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 11]
```diff
@@ -84,6 +105,29 @@ license signature. Those metrics are only exported on {{site.base_gateway}}.
 - **Number of Nginx timers** : A gauge metric that measures the total number of Nginx 
     timers, in Running or Pending state.
 
+Following metrics are disabled by default as it may create high cardinality of metrics and may
+cause performance issues:
+
+When `status_code_metrics` is set to true:
+- **Status codes**: HTTP status codes returned by Upstream services.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:112]:
```suggestion
- **Status codes**: HTTP status codes returned by upstream services.
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 12]
```diff
@@ -84,6 +105,29 @@ license signature. Those metrics are only exported on {{site.base_gateway}}.
...
+  These are available per service, across all services, and per route per consumer.
+
+When `lantency_metrics` is set to to true:
+- **Latencies Histograms**: Latency in ms, as measured at Kong:
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:116]:
```suggestion
- **Latencies Histograms**: Latency (in ms), as measured at Kong:
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 13]
```diff
@@ -84,6 +105,29 @@ license signature. Those metrics are only exported on {{site.base_gateway}}.
...
+   - **Request**: Total time taken by Kong and Upstream services to serve
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:117]:
```suggestion
   - **Request**: Total time taken by Kong and upstream services to serve
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 14]
```diff
@@ -84,6 +105,29 @@ license signature. Those metrics are only exported on {{site.base_gateway}}.
...
+     requests.
+   - **Kong**: Time taken for Kong to route a request and run all configured
+     plugins.
+   - **Upstream**: Time taken by the Upstream service to respond to requests.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:121]:
```suggestion
   - **Upstream**: Time taken by the upstream service to respond to requests.
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 15]
```diff
@@ -84,6 +105,29 @@ license signature. Those metrics are only exported on {{site.base_gateway}}.
...
+
+When `bandwidth_metrics` is set to true:
+- **Bandwidth**: Total Bandwidth (egress/ingress) flowing through Kong.
+  This metric is available per service and as a sum across all services.
+
+When `upstream_health_metrics` is set to true:
+- **Target Health**: The healthiness status (`healthchecks_off`, `healthy`, `unhealthy`, or `dns_error`) of Targets
+  belonging to a given Upstream as well as their subsystem (`http` or `stream`).
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:129]:
```suggestion
- **Target Health**: The healthiness status (`healthchecks_off`, `healthy`, `unhealthy`, or `dns_error`) of targets
  belonging to a given upstream as well as their subsystem (`http` or `stream`).
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 16]
```diff
@@ -210,6 +254,11 @@ allow access to the `/metrics` endpoint to Prometheus:
 
 ## Changelog
 
+### 3.0.x
+
+* Disable high cardinality metrics by default.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:259]:
```suggestion
* High cardinality metrics are now disabled by default.
```

File: app/_hub/kong-inc/prometheus/_index.md [hunk 17]
```diff
@@ -210,6 +254,11 @@ allow access to the `/metrics` endpoint to Prometheus:
...
+* Decreases performance penalty to proxy traffic when collecting metrics.
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/prometheus/_index.md:260]:
```suggestion
* Decreased performance penalty to proxy traffic when collecting metrics.
```

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]

**Review by cloudjumpercat — APPROVED** [review]

**Comment by hfukada** [comment]
shouldn't the prometheus 1.6.0 version be be moved to 3.0.0?

**Review by cloudjumpercat — APPROVED** [review]

**Pull request merged**